fvm/environment/programs.go

```diff
@@ -311,17 +327,16 @@ func (s *dependencyStack) push(loc common.Location) {
 func (s *dependencyStack) add(dependencies derived.ProgramDependencies) {
 	l := len(s.trackers)
 	if l == 0 {
-		// stack is empty.
-		// This is expected if loading a program that is already cached.
-		return
+		// This cannot happen, as the root of the stack is always present.
+		panic("Dependency stack unexpectedly empty")
```

Comment:
maybe switch to return failure after you've merged the other pr

Reply:
I will add another PR at the end of this PR chain for that.